routines.go

```diff
@@ -470,3 +499,165 @@ func WebsocketReconnect(ws *exchange.Websocket, verbose bool) {
+						bot.exchanges[x].GetName())
+					return
+				}
+
+				enabledCurrencies := bot.exchanges[x].GetEnabledCurrencies()
+				assetTypes, err := exchange.GetExchangeAssetTypes(exchangeName)
+				if err != nil {
+					log.Errorf("failed to get %s exchange asset types. Error: %s",
+						exchangeName,
+						err)
+					return
+				}
+
+				for y := range assetTypes {
+					for z := range enabledCurrencies {
+						fetchHistory(bot.exchanges[x],
+							enabledCurrencies[z],
+							assetTypes[y],
+							time.Unix(dbconfig.TimestampStart, 0),
+							time.Unix(dbconfig.TimestampEnd, 0))
+					}
+				}
+			}(x, &wg)
+		}
+		wg.Wait()
+		log.Debugln("All enabled currency platform trades fetched.")
+		time.Sleep(time.Minute * 5)
+	}
+}
+
+func fetchHistory(exch exchange.IBotExchange, p currency.Pair, assetType string, timeStart, timeEnd time.Time) {
+	if timeEnd.IsZero() {
+		timeEnd = time.Now()
+	}
+
+	// See if there is a last historic value entered into the database
+	lastTime, tradeID, err := bot.db.GetPlatformTradeLast(exch.GetName(),
+		p.String(),
+		assetType)
+	if err != nil {
+		if err.Error() != "sql: no rows in result set" {
+			log.Debugf("Getting last platform trade for %s %s %s and in time period [%s -> %s] error %s",
+				exch.GetName(),
+				p,
+				assetType,
+				timeStart,
+				timeEnd,
+				err)
+			return
+		}
+	}
+
+	if !lastTime.IsZero() {
+		if lastTime.After(timeEnd) || time.Now().Truncate(5*time.Minute).Unix() < lastTime.Unix() {
+			log.Debugf("Fetching platform rates finished for %s %s %s and in time period [%s -> %s]",
+				exch.GetName(),
+				p,
+				assetType,
+				timeStart,
+				timeEnd)
+			return
+		}
+	}
+
+	h, err := exch.GetPlatformHistory(p,
+		assetType,
+		lastTime,
+		tradeID)
+	if err != nil {
+		log.Errorf("No platform history returned for %s %s %s and in time period [%s -> %s] with error %s",
+			exch.GetName(),
+			p,
+			assetType,
+			timeStart,
+			timeEnd,
+			err)
+		return
+	}
+
+	if len(h) < 1 {
+		log.Errorf("No platform history returned for %s %s %s and in time period [%s -> %s]",
+			exch.GetName(),
+			p,
+			assetType,
+			timeStart,
+			timeEnd)
+		return
+	}
+
+	for i := range h {
+		err = bot.db.InsertPlatformTrade(h[i].TID,
+			h[i].Exchange,
+			p.String(),
+			assetType,
+			h[i].Type,
+			h[i].Amount,
+			h[i].Price,
+			h[i].Timestamp)
+		if err != nil {
+			if common.StringContains(err.Error(), "UNIQUE constraint failed") {
+				log.Errorf("%s %s %s Price: %.8f Amount: %.8f at [%s] error: %s",
+					h[i].Exchange,
+					p,
+					assetType,
+					h[i].Price,
+					h[i].Amount,
+					h[i].Timestamp,
+					err)
+				continue
+			}
+			log.Errorf("%s %s %s Price: %.8f Amount: %.8f at [%s] error: %s",
+				h[i].Exchange,
+				p,
+				assetType,
+				h[i].Price,
+				h[i].Amount,
+				h[i].Timestamp,
+				err)
+			return
+		}
+	}
+
+	printPlatformTradeSummary(&h,
+		exch.GetName(),
+		assetType,
+		p,
+		time.Time{},
+		time.Time{},
```

Comment:
This will create a new instance of time.Time which defaults to 01/01/1970 and not print the user selected time